test_conformance/images/clGetInfo/test_1D_buffer.cpp

```diff
@@ -0,0 +1,174 @@
+//
+// Copyright (c) 2023 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "../testBase.h"
+#include <CL/cl.h>
+
+extern int test_get_image_info_single(cl_context context,
+                                      image_descriptor *imageInfo, MTdata d,
+                                      cl_mem_flags flags, size_t row_pitch,
+                                      size_t slice_pitch);
+
+
+int test_get_image_info_1D_buffer(cl_device_id device, cl_context context,
+                                  cl_image_format *format, cl_mem_flags flags)
+{
+    size_t maxWidth;
+    cl_ulong maxAllocSize, memSize;
+    image_descriptor imageInfo = { 0 };
+    RandomSeed seed(gRandomSeed);
+    size_t pixelSize;
+
+    cl_mem_flags all_host_ptr_flags[5] = { flags, CL_MEM_ALLOC_HOST_PTR | flags,
+                                           CL_MEM_COPY_HOST_PTR | flags,
+                                           CL_MEM_ALLOC_HOST_PTR
+                                               | CL_MEM_COPY_HOST_PTR | flags,
+                                           CL_MEM_USE_HOST_PTR | flags };
+
```

Comment:
As per the spec section on creating images (USE_|ALLOC_|COPY_)_HOST_PTR flags are not valid for use when creating IMAGE1D_BUFFER objects so they might have to be handled differently in these tests. 

https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#_creating_image_objects

Reply:
I have updated the test accordingly